sys/drv_mc146818.c

```diff
@@ -0,0 +1,110 @@
+#include <isa.h>
+#include <common.h>
+#include <stdc.h>
+#include <vnode.h>
+#include <devfs.h>
+#include <errno.h>
+
+#define MC146818_ADDR 0x70
+#define MC146818_DATA 0x71
+
+#define MC146818_REG_SECOND 0x00
+#define MC146818_REG_MINUTE 0x02
+#define MC146818_REG_HOUR 0x04
+#define MC146818_REG_DAY 0x07
+#define MC146818_REG_MONTH 0x08
+#define MC146818_REG_YEAR 0x09
+#define MC146818_REG_B 0x0B
+
+#define MC146818_REG_B_BCD 0x04
+#define MC146818_REG_B_24H 0x02
+
+typedef struct mc146818_state { resource_t *io; } mc146818_state_t;
+
+static int mc146818_probe(device_t *dev) {
+  isa_device_t *isad = isa_device_of(dev);
+
+  if (!isad)
+    return 0;
+
+  /* TODO: Actually do some very basic test to see if the RTC is present. */
+
+  return 1;
+}
+
+static void mc146818_write_reg(mc146818_state_t *mc146818, uint8_t reg,
```

Comment:
That should be a part of a `bus_space` attached to resource delivered by PCI-ISA bridge. We could have exactly the same device, but accessible directly through physical memory address space, that why I'd like access methods to be abstracted away.